components/raftstore-v2/src/operation/command/admin/leader_transfer.rs

```diff
@@ -0,0 +1,419 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+
+use bytes::Bytes;
+use engine_traits::{KvEngine, RaftEngine, CF_LOCK};
+use fail::fail_point;
+use kvproto::{
+    disk_usage::DiskUsage,
+    metapb,
+    raft_cmdpb::{
+        AdminCmdType, AdminRequest, AdminResponse, CmdType, PutRequest, RaftCmdRequest, Request,
+        TransferLeaderRequest,
+    },
+};
+use parking_lot::RwLockWriteGuard;
+use raft::{eraftpb, ProgressState, Storage};
+use raftstore::{
+    store::{
+        fsm::new_admin_request, make_transfer_leader_response, metrics::PEER_ADMIN_CMD_COUNTER,
+        LocksStatus, Transport, TRANSFER_LEADER_COMMAND_REPLY_CTX,
+    },
+    Result,
+};
+use rand::prelude::SliceRandom;
+use slog::info;
+use txn_types::WriteBatchFlags;
+
+use super::AdminCmdResult;
+use crate::{
+    batch::StoreContext,
+    fsm::{ApplyResReporter, PeerFsmDelegate},
+    raft::{Apply, Peer},
+    router::{CmdResChannel, PeerMsg, PeerTick},
+};
+
+fn get_transfer_leader_cmd(msg: &RaftCmdRequest) -> Option<&TransferLeaderRequest> {
+    if !msg.has_admin_request() {
+        return None;
+    }
+    let req = msg.get_admin_request();
+    if !req.has_transfer_leader() {
+        return None;
+    }
+
+    Some(req.get_transfer_leader())
+}
+
+impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    /// Return true to if the transfer leader request is accepted.
```

Comment:
The "to" reads weird and should be removed. (I see it's copied from the original comment.)

Reply:
Ok, I see.